### PR TITLE
Add rule action contents analysis test

### PR DESCRIPTION
**Pull request by arturdryomov**
So much boilerplate but at least the documentation [is honest about it](https://docs.bazel.build/versions/master/skylark/testing.html):

> It requires a fair amount of boilerplate to set up the rules under test and the rules containing test assertions.

### CODE REVIEW

File: detekt/detekt_test.bzl [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+        "--report",
+        "xml:bazel-out/darwin-fastbuild/bin/detekt/test_target_detekt_report.xml",
+        "--report",
+        "html:bazel-out/darwin-fastbuild/bin/detekt/test_target_detekt_report.html",
+    ]
+
+    expected_inputs = [
+        "detekt/path A",
+        "detekt/path B",
+        "detekt/path C",
```
Comment (arturdryomov) [inline detekt/detekt_test.bzl]: Seems like paths are not escaped. Is it Bazel responsibility?

File: detekt/detekt_test.bzl [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        xml_report = True,
+    )
+
+    detekt_action_contents_test(
+        name = "detekt_action_contents_test",
+        target_under_test = ":test_target",
+    )
+
+def detekt_test_suite(name):
+    _test_detekt_action_contents()
```
Comment (arturdryomov) [inline detekt/detekt_test.bzl]: Actually I wanted to write another test on missing `srcs` but... we execute fine without it. I guess we should throw an error on missing `srcs` or something.

**Comment by arturdryomov** [comment]
@artem-zinnatullin, I think I won the path issue. Feels weird but works.

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -22,6 +22,8 @@ jobs:
         run: |
           curl --location --fail https://github.com/bazelbuild/bazelisk/releases/download/v1.1.0/bazelisk-linux-amd64 --output /tmp/bazelisk
           chmod +x /tmp/bazelisk
+      - name: Unit Tests
```
Comment (artem-zinnatullin) [inline .github/workflows/main.yml]: nit: Maybe "Starlark Unit Tests"? If we start adding more logic to wrapper/persistent worker it may become confusing

Comment (arturdryomov) [reply .github/workflows/main.yml]: Bazel calls these analysis tests, renamed and restructured to fit it.

Comment (artem-zinnatullin) [reply .github/workflows/main.yml]: 👍

**Pull request merged**